extensions/parquet/base/src/test/java/io/deephaven/parquet/base/util/CachedChannelProviderTest.java

```diff
@@ -154,6 +170,60 @@ public void testReuse10() throws IOException {
         Assert.assertEquals(closed.size(), 0);
     }
 
+    @Test
+    public void testInvalidation() throws IOException {
+        final SeekableChannelsProvider wrappedProvider = new TestChannelProvider();
+        final CachedChannelProvider firstCCP, secondCCP, thirdCCP;
+        SeekableByteChannel rc1, rc2, wc1;
+
+        firstCCP = new CachedChannelProvider(wrappedProvider, 100);
+        rc1 = firstCCP.getReadChannel("rc1");
+        wc1 = firstCCP.getWriteChannel("wc1", false);
+        // firstCCP -> rc1, wc1
+
+        secondCCP = new CachedChannelProvider(wrappedProvider, 100);
+        rc2 = secondCCP.getReadChannel("rc2");
+        // secondCCP -> rc2
+
+        CachedChannelProviderTracker.getInstance().invalidateChannels(new File("rc2"));
+        Assert.assertTrue(!firstCCP.invalid() &&
+                !((CachedChannelProvider.CachedChannel) rc1).invalid() &&
+                !((CachedChannelProvider.CachedChannel) wc1).invalid());
+        Assert.assertTrue(secondCCP.invalid() &&
+                ((CachedChannelProvider.CachedChannel) rc2).invalid());
+
+        thirdCCP = new CachedChannelProvider(wrappedProvider, 100);
+        wc1 = thirdCCP.getWriteChannel("wc1", true);
+        // thirdCCP -> wc1
+
+        CachedChannelProviderTracker.getInstance().invalidateChannels(new File("wc1"));
+        // Both firstCCP and thirdCCP, and their corresponding channels should be invalidated
+        Assert.assertTrue(firstCCP.invalid() &&
+                ((CachedChannelProvider.CachedChannel) rc1).invalid() &&
+                ((CachedChannelProvider.CachedChannel) wc1).invalid());
+        Assert.assertTrue(thirdCCP.invalid());
+    }
+
+    @Test
+    public void testTrackerCleanup() throws IOException {
+        final SeekableChannelsProvider wrappedProvider = new TestChannelProvider();
+        // Register cached channel providers with different files
+        for (int i = 0; i < CachedChannelProviderTracker.PROVIDER_MAP_CLEANUP_LIMIT - 1; i++) {
+            SeekableChannelsProvider ccp = new CachedChannelProvider(wrappedProvider, 100);
+            ccp.getReadChannel("rc" + i);
+        }
+        Assert.assertEquals(CachedChannelProviderTracker.getInstance().size(),
+                CachedChannelProviderTracker.PROVIDER_MAP_CLEANUP_LIMIT - 1);
+
+        // Trigger garbage collection to clear any old providers
```

Comment:
Can I add a test that relies on gc?
I saw some other test files doing this.

Reply:
I think this is legit, but I would get a more qualified second opinion.

Reply:
Maybe can force a reset on the first usage, but GC looks to be required to test this code.